Borrowed from coreboot 4.18 diff on nvramtool build recipes.
We can see here that the compiler to be used was rewritten to be gcc even if CC was specified from env variables.

--- ./util/nvramtool/Makefile.orig	2019-11-19 09:58:22.000000000 -0500
+++ ./util/nvramtool/Makefile	2022-12-08 10:56:56.199000000 -0500
@@ -16,7 +16,7 @@
 
 PROGRAM = nvramtool
 
-CC      = gcc
+CC     ?= gcc
 INSTALL = /usr/bin/env install
 PREFIX  = /usr/local
 CFLAGS  = -O2 -g -Wall -Wextra -Wmissing-prototypes -I. -DCMOS_HAL=1

--- ./util/nvramtool/Makefile.inc.orig	2019-11-19 09:58:22.000000000 -0500
+++ ./util/nvramtool/Makefile.inc	2022-12-08 11:00:50.297000000 -0500
@@ -36,8 +36,7 @@
 nvramtoolobj += win32mmap.o
 endif
 
-$(objutil)/nvramtool $(objutil)/nvramtool/accessors $(objutil)/nvramtool/cli:
-	mkdir -p $@
+additional-dirs += $(objutil)/nvramtool/accessors $(objutil)/nvramtool/cli
 
 $(objutil)/nvramtool/%.o: $(top)/util/nvramtool/%.c
 	printf "    HOSTCC     $(subst $(objutil)/,,$(@))\n"
